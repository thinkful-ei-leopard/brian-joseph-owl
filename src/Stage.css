.Stage-elements {
    background-color: grey;
}