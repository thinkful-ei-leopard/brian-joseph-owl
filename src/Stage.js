import React from 'react';
import './Stage.css';

function Stage(props) {
    // If the participant is not on the stage
    // exit the function
    if(props.onStage === false) {
        return null;
    }
    // If the participant is on the stage
    // create the element
    return ( 
        <div>
            <div className="Stage-user-info">
                <p>{props.name}</p>
                <div className="Stage-volume-indicator"></div>
                <i>Volume Icon here</i>
                <i>Magnifying Icon here</i>
            </div>
            <img src={props.avatar} alt="avatar for the person"/>
        </div>
    );
};

export default Stage;