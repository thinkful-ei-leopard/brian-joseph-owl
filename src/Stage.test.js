import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Stage from './Stage';

describe('<Stage />', () => {
  // smoke test
    it('renders without crashing', () => {
      const div = document.createElement('div');

      ReactDOM.render(<Stage/>, div);

      ReactDOM.unmountComponentAtNode(div);
    });
  //snapshot test
    it('renders this UI as expected', () => {
      const tree = renderer.create(<Stage />).toJSON();
      expect(tree).toMatchSnapshot();
    });
});
