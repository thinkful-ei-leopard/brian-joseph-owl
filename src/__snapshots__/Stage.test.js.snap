// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Stage /> renders this UI as expected 1`] = `<div />`;
